sources/api/schnauzer/src/v2/import/settings.rs

```diff
@@ -0,0 +1,319 @@
+//! This module contains the interface `SettingsResolver` for fetching settings from the Bottlerocket API.
+//!
+//! To do so we thinly wrap the apiclient to simplify retrieving JSON settings values.
+use super::as_std_err;
+use crate::v2::ExtensionRequirement;
+use async_trait::async_trait;
+use constants::API_SETTINGS_URI;
+use serde_json::{json, Map, Value};
+use snafu::{ensure, OptionExt, ResultExt};
+use std::path::{Path, PathBuf};
+
+const SETTINGS_API_URI: &str = "/";
+
+/// An interface which abstracts away the fetching of Bottlerocket settings for template rendering.
+#[async_trait]
+pub trait SettingsResolver {
+    async fn fetch_settings<I>(
+        &self,
+        extension_requirements: I,
+    ) -> std::result::Result<serde_json::Value, Box<dyn std::error::Error>>
+    where
+        I: Iterator<Item = ExtensionRequirement> + Send;
+}
+
+/// `SettingsResolver` implementation that fetches settings from the Bottlerocket API.
+#[derive(Debug, Clone)]
+pub struct BottlerocketSettingsResolver {
+    pub api_socket: PathBuf,
+}
+
+impl BottlerocketSettingsResolver {
+    pub fn new(api_socket: PathBuf) -> Self {
+        Self { api_socket }
+    }
+}
+
+impl Default for BottlerocketSettingsResolver {
+    fn default() -> Self {
+        Self {
+            api_socket: constants::API_SOCKET.into(),
+        }
+    }
+}
+
+#[async_trait]
+impl SettingsResolver for BottlerocketSettingsResolver {
+    /// Fetches requested settings from the Bottlerocket API.
+    async fn fetch_settings<I>(
+        &self,
+        extension_requirements: I,
+    ) -> std::result::Result<serde_json::Value, Box<dyn std::error::Error>>
+    where
+        I: Iterator<Item = ExtensionRequirement> + Send,
+    {
+        // TODO: Modify this to use per-setting requests in the future.
+        let all_settings = Self::get_settings_json(&self.api_socket)
+            .await?
+            .as_object()
+            .cloned()
+            .context(error::NonJSONObjectSnafu {
+                key: "settings".to_string(),
+            })
+            .map_err(as_std_err)?;
```

Comment:
How expensive is this query and how many times do we invoke it on each boot? Can we safely cache the response and use it for multiple templates?

Reply:
It's definitely a significant increase in API calls compared to what we're doing now (once-per-thar-be-settings, instead of once-per-template). I'll modify this to cache the response, although we'll need to put some effort into parallelizing the apiserver and performing these queries in parallel when settings extensions are merged, since we can't fetch all values at once in that world.